### PR TITLE
feat(OM-054): ONNX and WebGPU export targets

### LINKED ISSUE #175: EPIC: Add ONNX and WebGPU export targets to model conversion
## Summary
ONNX/WebGPU is 'the one genuinely-missing export target' (section 4.2), a v1.7 headline (section 8.2), and a KPI (section 8.5). Today mlx/convert.py and coreml/convert.py emit only MLX/CoreML variants; OM-013 adds push_to_hub for those, OM-045 skips convert.py, and OM-032 only computes quant G4. There is no ONNX/WebGPU export path, so OpenMed cannot ship browser/cross-platform runtime artifacts. Decompose before starting.

## Scope
- [ ] Decompose before starting into: (a) extend the convert pipeline (mlx/convert.py + coreml/convert.py or a new openmed/onnx/convert.py) to export ONNX and WebGPU-targeted artifacts from a source checkpoint; (b) emit the new formats into the manifest formats[] for the published repo (feeds OM-013 push_to_hub); (c) gate the exported artifacts via the G4 quant-delta + G5 tier-fit checks so an ONNX/WebGPU variant cannot ship below its FP parent's recall/SLO; (d) document the export and runtime-load path.
- [ ] Reuse the existing convert + HF publish + gate machinery; do not rebuild the conversion pipeline.

## Acceptance criteria
- [ ] Decompose before starting: epic is broken into independently-mergeable S/M/L issues before implementation begins.
- [ ] The convert pipeline produces an ONNX artifact and a WebGPU-targeted artifact from a source checkpoint; both appear in the manifest formats[] for the published repo.
- [ ] Exported variants are gated through G4 (quant delta vs FP parent) and G5 (tier fit) and cannot ship when below the threshold.
- [ ] The export and runtime-load path is documented.
- [ ] test suite green: .venv/bin/python -m pytest tests/ -q

## Out of scope
- The HF push_to_hub step itself (OM-013, consumed).
- The CI fail-closed wiring and quant-delta evaluator (OM-032, consumed).
- Multimodal redaction (OM-045).

## Files
- openmed/onnx/convert.py
- openmed/mlx/convert.py
- openmed/coreml/convert.py
- docs/export-onnx-webgpu.md

---
Task: OM-054  ·  Milestone: v1.7  ·  Priority: P1  ·  Size: L
Depends on: OM-013, OM-032  ·  Blocks: —
Roadmap: section 4.2 (export targets), section 8.2 (v1.7 headline), section 8.5 (KPI)
Spec: PLANS/V2/EXECUTION/tasks/OM-054.md

### PR DESCRIPTION
**Pull request by matdou**
## Description

Adds an ONNX export pipeline and a WebGPU-targeted fp16 artifact to OpenMed, so token-classification models can run in browsers and cross-platform runtimes. Implements the core of OM-054 decomposed into two independently-shippable pieces.

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Test addition/improvement

## Changes Made

- `openmed/onnx/convert.py` — exports any HF token-classification model to ONNX; handles torch 2.x dynamo exporter correctly (`torch.export.Dim`, opset 18, batch=2 trace to avoid BERT position-embedding specialisation); CLI mirrors `coreml/convert.py`
- `openmed/onnx/webgpu.py` — fp16 post-export conversion via `onnxruntime.transformers.float16` with `keep_io_types=True` (int64 inputs, float32 outputs preserved for WebGPU EP); standalone CLI + `--webgpu` flag on the convert CLI
- `openmed/eval/gates.py` — fail-closed stub for G4 `quant_delta` and G5 `tier_fit` with `TODO(OM-032)`; raises `NotImplementedError` until OM-032 lands
- `openmed/mlx/artifact.py` — `write_manifest()` now also emits `formats[]` alongside the existing scalar `format` (additive, no reader breakage)
- `pyproject.toml` — `[onnx]` optional dependency group: `onnx>=1.14`, `onnxruntime>=1.16`, `torch>=2.0`, `transformers>=4.50`

## Testing

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have tested this change with different models/inputs

T1 parity: torch vs ONNX CPU EP max abs diff `7.45e-09`, 100% argmax agreement on batch 1–8 × seq 8–64. Verified end-to-end with `hf-internal-testing/tiny-bert-for-token-classification`.

```
uv run pytest tests/unit/onnx/ -v          # T1 parity, T3 manifest, T5 WebGPU
uv run pytest tests/ -q --ignore=tests/integration  # full suite: 1259 passed
cd tests/node && npm install && npm test   # T4 Node WASM EP (generate fixtures first)
WEBGPU_SMOKE=1 node tests/node/test_onnx_webgpu.mjs  # T6 headed-Chrome WebGPU smoke
```

## Documentation

- [x] I have added docstrings to new functions/classes

## Code Quality

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

## Dependencies

- [x] I have added new dependencies and they are justified because: `onnx`, `onnxruntime`, and `transformers` are required for the ONNX export path; all gated behind `pip install openmed[onnx]` so existing installs are unaffected

## Checklist

- [x] I have read the contributing guidelines
- [x] My commits have clear, descriptive messages

## Related Issues

Closes #175

### CODE REVIEW

**Comment by maziyarpanahi** [comment]
Thank you @matdou for your great contribution. I will review this today and we'll get this merged 🎉

**Comment by maziyarpanahi** [comment]
Superseded by #383, which uses the origin-owned branch for #175 after the prerequisites were merged.
